### ISSUE TITLE
error pulling image: Cannot overwrite digest

### ISSUE THREAD

**Issue by kargakis** [openshift/release#1842]
Just saw the following error in the log of one of our jobs
```
2018/10/08 15:13:21 Build e2e-tests failed, printing logs:
Pulling image "docker-registry.default.svc:5000/ci-op-j221slnq/pipeline@sha256:ccce6dfae8c79adcefe9c65e13cf3eaaa871daa5610a4f295c1b5de86b564e71" ...
error: error pulling image docker-registry.default.svc:5000/ci-op-j221slnq/pipeline@sha256:ccce6dfae8c79adcefe9c65e13cf3eaaa871daa5610a4f295c1b5de86b564e71: Cannot overwrite digest sha256:ccce6dfae8c79adcefe9c65e13cf3eaaa871daa5610a4f295c1b5de86b564e71
```
https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_openshift-azure/569/pull-ci-openshift-openshift-azure-master-images/94/build-log.txt

Not sure what the issue is there

[JOB SPEC](https://github.com/openshift/release/blob/fe2052b006680153cf9363adf1ec32dc8f66d923/ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-master-presubmits.yaml#L182) + [CI-OPERATOR CONFIG](https://github.com/openshift/release/blob/master/ci-operator/config/openshift/openshift-azure/openshift-openshift-azure-master.yaml)

/kind bug

**Comment by bparees** [comment]
neither.  it's a docker runtime issue.  @mrunalp's team has a fix but i don't think our AMIs include it yet.

**Comment by stevekuznetsov** [comment]
Ah, we would need to upgrade our nodes in api.ci for that

/cc @smarterclayton

**Comment by legionus** [comment]
This is a client error. Not registry server.

https://github.com/openshift/origin/blob/master/vendor/github.com/docker/docker/reference/store.go#L154

**Comment by smarterclayton** [comment]
We should have fixed this a month ago (I pulled new rpms and rebuilt images).  If this is still happening the fix wasn’t successful.

**Comment by bparees** [comment]
This came up last week and I pinged mrunal and Miroslav who confirmed at
least in the job I was looking at the rpm was not new enough to have the
fix.


Ben Parees | OpenShift

**Comment by stevekuznetsov** [comment]
What job was it? This is the RPM on our cluster nodes, how did you check that version?

**Comment by bparees** [comment]
https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/batch/test_pull_request_origin_extended_image_ecosystem/1246/

see the discussion in forum-runtimes (you can search on that link to find the discussion)

So that's an AWS-based job, so different base.

Here's what miroslav said though:  "Random grepping in the output suggests the failing version is docker-1.13.1-63.git94f4240.el7 , the fix is in docker-1.13.1-75.git8633870.el7_5 ."

**Comment by smarterclayton** [comment]
Not positive, but I think we have 75 on the cluster and are seeing failures
now.  Is the update in dockertested?

https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/21244/pull-ci-openshift-origin-master-e2e-gcp/695/?log#log


On Oct 8, 2018, at 5:20 PM, Ben Parees <notifications@github.com> wrote:

https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/batch/test_pull_request_origin_extended_image_ecosystem/1246/

see the discussion in forum-runtimes (you can search on that link to find
the discussion)

So that's an AWS-based job, so different base.

Here's what miroslav said though: "Random grepping in the output suggests
the failing version is docker-1.13.1-63.git94f4240.el7 , the fix is in
docker-1.13.1-75.git8633870.el7_5 ."

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/release/issues/1842#issuecomment-427982953>,
or mute the thread

**Comment by mrunalp** [comment]
@smarterclayton Yes, I can see in the logs here https://storage.googleapis.com/origin-ci-test/pr-logs/pull/21315/pull-ci-openshift-origin-master-e2e-gcp/966/build-log.txt that we are on build 75.
@mtrmac @runcom Can you take a look?

**Comment by mtrmac** [comment]
I’m afraid the “Fixed in version” field of https://bugzilla.redhat.com/show_bug.cgi?id=1628262 is incorrect; in fact none of the el7 packages contain the fix AFAICS, because https://github.com/projectatomic/docker/pull/321 was merged into the `docker-1.13.1` branch, but not the `docker-1.13.1-rhel` branch.

@runcom/@lsm5 What is the process to get https://github.com/projectatomic/docker/pull/321 into the relevant `docker` RPMs? (`docker-1.13.1-rhel` seems not to use a PR workflow, or at least not consistently.) Also, https://bugzilla.redhat.com/show_bug.cgi?id=1628262 might need updating.

**Comment by runcom** [comment]
I'm backporting it now.

**Comment by runcom** [comment]
just backported https://github.com/projectatomic/docker/commit/9e822128cab3254384de074658d0a53044a38f09

**Comment by lsm5** [comment]
On Sun, Oct 21, 2018 at 08:20:32AM -0700, Clayton Coleman wrote:
> I’ll respin the cluster image as soon as we have an rpm in dockertested
> 

Done, https://bugzilla.redhat.com/show_bug.cgi?id=1628262#c10

-- 
Lokesh
IRC, GitHub: lsm5
GPG: 0xC7C3A0DD
https://keybase.io/lsm5

**Comment by smarterclayton** [comment]
The cluster has been updated.  New nodes will be rotated in with 82,
although old nodes may still stick around for the next day or so.

**Comment by dmage** [comment]
https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_cluster-image-registry-operator/33/pull-ci-openshift-cluster-image-registry-operator-master-images/95/
https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_cluster-image-registry-operator/33/pull-ci-openshift-cluster-image-registry-operator-master-e2e-aws/44
I hope it's just an old node.
